specs/WebMessageObjects.md

```diff
@@ -157,11 +234,165 @@ interface ICoreWebView2WebMessageReceivedEventArgs2 : ICoreWebView2WebMessageRec
+  COREWEBVIEW2_FILE_SYSTEM_HANDLE_KIND_DIRECTORY
+} COREWEBVIEW2_FILE_SYSTEM_HANDLE_KIND;
+
+[v1_enum]
+typedef enum COREWEBVIEW2_FILE_SYSTEM_HANDLE_PERMISSION {
+  /// Read-only permission for FileSystemHandle
+  COREWEBVIEW2_FILE_SYSTEM_HANDLE_PERMISSION_READONLY,
+  /// Read and write permissions for FileSystemHandle
+  COREWEBVIEW2_FILE_SYSTEM_HANDLE_PERMISSION_READWRITE
+} COREWEBVIEW2_FILE_SYSTEM_HANDLE_PERMISSION;
+
+[uuid(0ecf4d7d-bbf6-4320-930e-82ff6cf2d8dc), object, pointer_default(unique)]
+interface ICoreWebView2FileSystemHandle : IUnknown {
+  /// The Kind of the FileSystemHandle. It can either be a file or a directory.
+  [propget] HRESULT Kind([out, retval] COREWEBVIEW2_FILE_SYSTEM_HANDLE_KIND* kind);
+
+  /// The path to the FileSystemHandle.
+  [propget] HRESULT Path([out, retval] LPWSTR* path);
+
+  /// The permissions granted to the FileSystemHandle.
+  [propget] HRESULT Permission([out, retval] COREWEBVIEW2_FILE_SYSTEM_HANDLE_PERMISSION* kind);
+}
+
+// Generic container of COM objects to pass to WebView2
+[uuid(d2fba648-109f-401c-b701-44fd3ddfb440), object, pointer_default(unique)]
+interface ICoreWebView2ObjectCollection : ICoreWebView2ObjectCollectionView {
+  /// Removes the object at the specified index.
+  HRESULT RemoveValueAtIndex([in] UINT32 index);
+
+  /// Inserts the object at the specified index.
+  HRESULT InsertValueAtIndex(
+      [in] UINT32 index,
+      [in] IUnknown* value);
+}
+
+[uuid(afae6f48-60d6-4b95-8b81-6f60d9c70f0b), object, pointer_default(unique)]
+interface ICoreWebView2Environment14 : ICoreWebView2Environment13 {
+  /// Create a ICoreWebView2FileSystemHandle from a path. The `path` is the path pointed by the
+  /// file and must be a syntactically correct full path but it is not checked here whether it currently
+  /// points to a file or a directory. If an invalid path is passed, an E_INVALIDARG will be returned
+  /// and the object will fail to create.
+  ///
+  /// If the `kind` is COREWEBVIEW2_FILE_SYSTEM_HANDLE_KIND_FILE, representation of a Web
+  /// [FileSystemFileHandle](https://developer.mozilla.org/docs/Web/API/FileSystemFileHandle)
+  /// will be created. Once the object is passed to web content, if the content is attempting a read,
+  /// the file must be existing and available to read similar to a file chosen by
+  /// [open file picker](https://developer.mozilla.org/docs/Web/API/Window/showOpenFilePicker),
+  /// otherwise the read operation will throw a DOM exception. For write operations the file does not
+  /// need to exist as `FileSystemFileHandle` will behave more like a file path chosen by 
+  /// [save file picker](https://developer.mozilla.org/docs/Web/API/Window/showSaveFilePicker)
+  /// and will create or overwrite the file, but the parent directory structure pointed 
+  /// by the file must exist and an existing file must be available to write and delete
+  /// or the write operation will throw a DOM exception.

```

Comment:
Can CoreWebView2 be used from UWP? (I assume not. Otherwise we have even bigger problems with paths and security.)